src/tsickle.ts

```diff
@@ -1452,30 +1457,30 @@ class Annotator extends ClosureRewriter {
     this.emit(`${namespace.join('.')};\n`);
   }
 
+  private typedefCounter = 1;
+
   private visitTypeAlias(node: ts.TypeAliasDeclaration) {
     // If the type is also defined as a value, skip emitting it. Closure collapses type & value
     // namespaces, the two emits would conflict if tsickle emitted both.
     const sym = this.mustGetSymbolAtLocation(node.name);
     if (sym.flags & ts.SymbolFlags.Value) return;
+    // Type aliases are always emitted as the resolved underlying type, so there is no need to emit
+    // anything, except for exported types.
+    if (!hasModifierFlag(node, ts.ModifierFlags.Export) || !this.host.googmodule) return;
+
+    const typeName = node.name.getText();
 
-    // Write a Closure typedef, which involves an unused "var" declaration.
+    // Blacklist any type parameters, Closure does not support type aliases with type parameters.
     this.newTypeTranslator(node).blacklistTypeParameters(
         this.symbolsToAliasedNames, node.typeParameters);
-
-    const typeStr =
-        this.host.untyped ? '?' : this.typeToClosure(node, undefined, true /* resolveAlias */);
-    const typeName = node.name.getText();
-    this.emit(`\n/** @typedef {${typeStr}} */\n`);
-    this.emit(`var ${typeName};\n`);
+    const typeStr = this.host.untyped ? '?' : this.typeToClosure(node, undefined);
     // In the case of an export, we cannot emit a `export var foo;` because TypeScript drops exports
     // that are never assigned values (and Closure requires us to not assign values to typedef
     // exports).
     // tsickle must also still emit the `var` line above so that the type can be used within the
     // local module scope (Closure does not allow refering to "exports.Foo" within a module).
-    // With that, emit an additional "export {foo};" to export the type.
-    if (hasModifierFlag(node, ts.ModifierFlags.Export)) {
-      this.emit(`export {${typeName}};\n`);
-    }
+    // With that, emit an additional "export {foo_mangled as foo};" to export the type.
```

Comment:
This comment ... is left from the previous change I think?

Reply:
Yeah, fixed.